lib/spack/spack/test/cmd/compiler_command.py

```diff
@@ -34,7 +34,7 @@ def test_compiler_find_without_paths(no_compilers_yaml, working_env, tmpdir):
 """)
         os.chmod('gcc', 0o700)
 
-    os.environ['PATH'] = str(tmpdir)
+    os.environ['PATH'] = '/usr/sbin:' + str(tmpdir)
```

Comment:
Is this not also an issue with the detection logic?

@alalazo: Should we do something similar in `llnl.util.cpu.detect.sysctl()`?

Reply:
I don't think it's an issue because it's only in this test that we completely blow away the entire `PATH`. Normally we want to keep the `PATH` so we can find compilers in it. Although I'm not sure how `spack compiler add <path>` works.

Reply:
It looks like it's always either in `/sbin` or `/usr/sbin`.  Users do all sorts of things to their paths -- I think it would be good to have the detection logic find this if it's not in `PATH`